src/packages/DAO/elections.dao.ts

```diff
@@ -53,6 +53,23 @@ export async function startElection(formData: FormData) {
   revalidatePath("/");
 }
 
+const VALUES = ["NOT_STARTED", "ONGOING", "FINISHED"] as const;
+
+export async function addElection(formData: FormData) {
+  const electionName = String(formData.get("electionName"));
```

Comment:
If you check the file `src/packages/DTO/elections.dto.ts`, you will see that we are defining zod validators there, and, in the past, I already created a validator like:

```tsx
export const createElectionValidator = z.object({
  name: z.string(),
  electionDay: z.date(),
});
```

Update the method there, to your data format, notice that I've created `createElectionValidator` but it's not being used currently, so please, do changes as you like, import here, and validate your data as an object.

Reply:
At the `elections.dto.ts file`, you can define your validator as in:

```tsx
import { ElectionStatus } from "@prisma/client";
import { z } from "zod";

export const createElectionValidator = z.object({
  name: z.string().min(1),
  status: z.nativeEnum(ElectionStatus),
});
```

**PLEASE NOTE** For the ENUM definition, I'm using the actual ENUM that's used by Prisma to define this column in the database.

Meaning, I'm avoiding code duplication. If someone updates this ENUM in the database I don't need to also update it on multiple points in my Application.

Reply:
To finish, this is a version of your method using this new validator that I've written above:

```tsx
export async function addElection(formData: FormData) {
  const data = {
    name: formData.get("electionName"),
    status: formData.get("electionStatus"),
  };

  const parsedData = createElectionValidator.parse(data);

  return await prisma.election.create({
    data: parsedData,
  });
}
```

Reply:
Let me know what do you think!